pocs/scheduler/constraint.py

```diff
@@ -159,3 +165,160 @@ def get_score(self, time, observer, observation, **kwargs):
 
     def __str__(self):
         return "Moon Avoidance"
+
+
+class Horizon(BaseConstraint):
+
+    """ Implements horizon and obstruction limits"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.obstruction_points = []
+
+        print("Horizon.__init__")
+        for i in self.config:
+            print(i, self.config[i])
+
+    def set_obstruction_points(self, op):
+        self.obstruction_points = op
+
+    def process_image(self, image_filename):
+        """
+        Process the horizon_image to generate the obstruction_points list
+        Segment regions of high contrast using scikit image
+        Image Segmentation with Watershed Algorithm
+        bottom_left is a tuple, top_right is a tuple, each tuple has az, el
+        to allow for incomplete horizon images
+
+        Note: 
+            Incomplete method, further work to be done to automate the horizon limits
+
+        Args:
+            image_filename (file): The horizon panorama image to be processed
+        """
+
+        from skimage.io import imread
+        from skimage.filters import threshold_otsu
+        from skimage import feature
+        import numpy as np
+
+        image = imread(image_filename, flatten=True)
+        thresh = threshold_otsu(image)
+        binary = image > thresh
+
+        # Compute the Canny filter
+        edges1 = feature.canny(binary, low_threshold=0.1, high_threshold=0.5)
+
+        # Turn into array
+        np.set_printoptions(threshold=np.nan)
+        print(edges1.astype(np.float))
+
+    def get_config_coords(self):
```

Comment:
This is reading the config file but not necessarily "getting" them since it doesn't return anything. I would rename to `lookup_config_coords` or something.